dashboard/app/controllers/regional_partners_controller.rb

```diff
@@ -74,9 +74,7 @@ def remove_program_manager
 
   # GET /regional_partners/:id/search_program_manager
   def search_program_manager
-    search_term = params[:search_term]
-    teachers = restricted_users.where(user_type: 'teacher')
-    @users = teachers.where("email LIKE :partial_email", {partial_email: "%#{search_term}%"}).or(teachers.where("name LIKE :partial_name", {partial_name: "%#{search_term}%"}))
+    @users = restricted_users.where(user_type: 'teacher', hashed_email: restricted_users.hash_email(params[:search_term]))
```

Comment:
nit: for readability, split up this line:
```ruby
@users = restricted_users.where(
  user_type: 'teacher', 
  hashed_email: restricted_users.hash_email(params[:search_term])
)
```

Also note you could use `User.hash_email`, which is a bit more concise and functionally equivalent. No strong preference here.